### PR TITLE
GA Migration RAT Spotless Whitespace PreCommit

### PR DESCRIPTION
**Pull request by fernando-wizeline**
As part of the migration of Precommit and Postcommit Jobs from Jenkins to GA in self-hosted runners, this PR contains:

Migrated workflow [job-precommit-rat.yml](https://github.com/fernando-wizeline/beam/blob/gam-precommit-rat-spotless-whitespace/.github/workflows/job-precommit-rat.yml)
Migrated workflow [job-precommit-spotless.yml](https://github.com/fernando-wizeline/beam/blob/gam-precommit-rat-spotless-whitespace/.github/workflows/job-precommit-spotless.yml)
Migrated workflow [job-precommit-whitespace.yml](https://github.com/fernando-wizeline/beam/blob/gam-precommit-rat-spotless-whitespace/.github/workflows/job-precommit-whitespace.yml)

The migrated workflow was removed from [.test-infra/jenkins/README.md](https://github.com/fernando-wizeline/beam/blob/gam-precommit-rat-spotless-whitespace/.test-infra/jenkins/README.md) and added to [CI.md](https://github.com/fernando-wizeline/beam/blob/gam-precommit-rat-spotless-whitespace/CI.md)

DO NOT MERGE until the effort to use self-hosted runners is completed https://github.com/apache/beam/pull/22703
------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by fernando-wizeline** [comment]
Hi @damccorm!
We're facing a minor inconvenience regarding the checkstyle report created by the spotless task. In jenkins, that task creates a report like the [following](https://ci-beam.apache.org/job/beam_PreCommit_Spotless_Commit/22573/checkstyle/)
There are a couple of unofficial plugins that generate a really short status section within github, although not that different from the one already provided by github.
My question is: would it be ok if we skip the report generation for the time being? Please let me know what you think about this.
Thanks!

**Comment by damccorm** [comment]
> My question is: would it be ok if we skip the report generation for the time being? Please let me know what you think about this.

What does the current experience look like when there is a violation? I'm fine not generating the report as long as there's a good way to figure out what violation caused a workflow to fail

**Comment by fernando-wizeline** [comment]
> > My question is: would it be ok if we skip the report generation for the time being? Please let me know what you think about this.
> 
> What does the current experience look like when there is a violation? I'm fine not generating the report as long as there's a good way to figure out what violation caused a workflow to fail

Hi Danny!
Currently the errors are presented in the results window of the github job right at the end, with all the info as to why they failed the validation.

**Comment by damccorm** [comment]
> Currently the errors are presented in the results window of the github job right at the end, with all the info as to why they failed the validation.

That sounds fine to me - depending on how good they are we might not need to come back to report generation at all. Any chance you could share a screenshot of an example?

**Comment by fernando-wizeline** [comment]
> > Currently the errors are presented in the results window of the github job right at the end, with all the info as to why they failed the validation.
> 
> That sounds fine to me - depending on how good they are we might not need to come back to report generation at all. Any chance you could share a screenshot of an example?

Totally!
This is the screenshot:
Screen Shot 2022-09-15 at 10 32 17

and the link to the results windows is this: https://github.com/fernando-wizeline/beam/actions/runs/3061434614/jobs/4941196972

**Comment by damccorm** [comment]
Thanks for sharing - lets definitely skip this for now. Since its a little tough to read that output, I'd like to at least explore options for a nicer report eventually, but it definitely should not block us moving this over to GHA.

**Comment by fernando-wizeline** [comment]
> Thanks for sharing - lets definitely skip this for now. Since its a little tough to read that output, I'd like to at least explore options for a nicer report eventually, but it definitely should not block us moving this over to GHA.

Great! Sounds like a plan. We'll move forward with this and explore alternatives for later implementation.
Thanks again Danny!

**Review by benWize — APPROVED** [review]
LGTM!

**Comment by fernando-wizeline** [comment]
Run Spotless PreCommit

File: sdks/python/test-suites/tox/py38/build.gradle [hunk 1]
```diff
@@ -112,6 +112,8 @@ task archiveFilesToLint(type: Zip) {
     include "**/*.md"
     include "**/build.gradle"
     exclude "**/node_modules/*"
+    exclude ".github/*"
+    exclude "website/*"
```
Comment (damccorm) [inline sdks/python/test-suites/tox/py38/build.gradle:116]: Did you mean to include this? Why do we need it?

**Review by damccorm — COMMENTED** [review]
Just had one question. Could you please file an issue for the more nicely formatted report as well?

File: sdks/python/test-suites/tox/py38/build.gradle (hunk 1 above)
Comment (fernando-wizeline) [reply sdks/python/test-suites/tox/py38/build.gradle:116]: Yeah, they're being picked up by the py38 process (not sure why) so I had to manually exclude them from the paths.

**Comment by fernando-wizeline** [comment]
> In jenkins, that task creates a report like the [following](https://ci-beam.apache.org/job/beam_PreCommit_Spotless_Commit/22573/checkstyle/)
> There are a couple of unofficial plugins that generate a really short status section within github, although not that different from the one already provided by github.

Issue filed here: https://github.com/apache/beam/issues/23466

**Review by damccorm — APPROVED** [review]
Thanks!

**Comment by damccorm** [comment]
Closing as stale

**Pull request closed without merging**